sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -468,13 +469,65 @@ abstract class StringPredicate extends BinaryExpression
   override def toString: String = s"$nodeName($left, $right)"
 }
 
-/**
- * A function that returns true if the string `left` contains the string `right`.
- */
+trait StringBinaryPredicateExpressionBuilderBase extends ExpressionBuilder {
+  override def build(funcName: String, expressions: Seq[Expression]): Expression = {
+    val numArgs = expressions.length
+    if (numArgs == 2) {
+      if (expressions(0).dataType == BinaryType && expressions(1).dataType == BinaryType) {
+        BinaryStringPredicate(funcName, expressions(0), expressions(1))
+      } else {
+        createStringPredicate(expressions(0), expressions(1))
+      }
+    } else {
+      throw QueryCompilationErrors.invalidFunctionArgumentNumberError(Seq(2), funcName, numArgs)
+    }
+  }
+
+  protected def funcName: String
+  protected def createStringPredicate(left: Expression, right: Expression): Expression
+}
+
+object StringPredicate {
+  def unapply(expr: Expression): Option[Expression] = expr match {
+    case _: StringPredicate => Some(expr)
+    case s @ StaticInvoke(clz, _, "contains" | "startsWith" | "endsWith", Seq(_, _), _, _, _, _)
+      if clz == classOf[ByteArrayMethods] => Some(s)
+    case _ => None
+  }
+}
+
+case class BinaryStringPredicate(funcName: String, left: Expression, right: Expression)
+  extends RuntimeReplaceable with ImplicitCastInputTypes {
+
+  private lazy val realFuncName = funcName match {
```

Comment:
If `funcName` isn't real, how about we just name it `prettyName`? Then we don't need to override `prettyName` below.

Reply:
> If `funcName` isn't real, how about we just name it `prettyName`? Then we don't need to override `prettyName` below.

Yea, for this case, prettyName if func name